### ISSUE TITLE
Issues on AOT:  'NgForageModule' references 'InstanceFactory'

### ISSUE THREAD

**Issue by CanKattwinkel** [Alorel/ngforage#56]
1. create a new project with angular cli
1. ` yarn add localforage@^1.5.0 ngforage@^2.0.0`
1. import using `      NgForageModule.forRoot()`
1. add constructor to AppModule:
```
    public constructor(ngfConfig: NgForageConfig) {
      ngfConfig.configure({
        name: 'MyApp',
        driver: [ // defaults to indexedDB -> webSQL -> localStorage -> sessionStorage
          NgForageConfig.DRIVER_INDEXEDDB,
          NgForageConfig.DRIVER_LOCALSTORAGE
        ]
      });
    }
```

build with aot:

```
ERROR in Error during template compile of 'AppModule'
  Could not resolve ../config/NgForageConfig.service relative to /home/can/projects/kill/ng-forage-test/node_modules/ngforage/ngforage.d.ts. in 'NgForageModule'
    'NgForageModule' references 'InstanceFactory'
      'InstanceFactory' contains the error at ../ngforage/ngforage.ts(50,12).

```

Versions:

```
Angular CLI: 1.7.3
Node: 7.7.3
OS: linux x64
Angular: 5.2.9
... animations, common, compiler, compiler-cli, core, forms
... http, language-service, platform-browser
... platform-browser-dynamic, router

@angular/cli: 1.7.3
@angular-devkit/build-optimizer: 0.3.2
@angular-devkit/core: 0.3.2
@angular-devkit/schematics: 0.3.2
@ngtools/json-schema: 1.2.0
@ngtools/webpack: 1.10.2
@schematics/angular: 0.3.2
@schematics/package-update: 0.3.2
typescript: 2.5.3
webpack: 3.11.0


ngforage:  2.0.2
localforage  1.7.1,
```

**Comment by Alorel** [comment]
Hey, I just noticed this as well and am trying to figure out what's wrong - I'm using the same version of ngtools/webpack & Angular + webpack 4. Oddly enough it compiles fine in JiT mode in another project and in AoT mode for the demo site (hence the CI passing).

If you have any idea on what might be causing this please let me know - I'm completely clueless at the moment.

**Comment by Alorel** [comment]
@zaalbarxx @CanKattwinkel 

This should now be fixed in versions 2.0.3 and 3.0.4 respectively - thanks a lot for the report :slightly_smiling_face:
